frontend/src/features/rubricBuilder/CSVUpload.tsx

```diff
@@ -1,71 +1,126 @@
-import React from "react";
-import Papa from "papaparse";
-import { CSVRow } from "@local_types";
+import React, { useState } from "react";
+import { importCsv } from "@utils";
+import { Dialog } from "@components";
+
+import { Criteria, Rubric } from "palette-types";
+import { v4 as uuid } from "uuid";
 
 interface CSVUploadProps {
-  onDataChange: (data: CSVRow[]) => void;
-  closeImportCard: () => void; // callback to close the import card
+  rubric: Rubric | undefined;
+  setRubric: React.Dispatch<React.SetStateAction<Rubric | undefined>>;
 }
 
-export const CSVUpload: React.FC<CSVUploadProps> = ({
-  onDataChange,
-  closeImportCard,
-}: CSVUploadProps) => {
-  const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    const file = event.target.files?.[0];
-    if (!file) return;
+export const CSVUpload: React.FC<CSVUploadProps> = ({ setRubric }) => {
+  const [showVersionModal, setShowVersionModal] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
 
-    const fileExtension = file.name.split(".").pop()?.toLowerCase();
+  const closeErrorDialog = () => setErrorMessage(null);
 
-    if (fileExtension === "csv") {
-      parseCSV(file);
-    } else {
+  const handleFileChange = (
+    file: File,
+    version: "versionOne" | "versionTwo",
+  ) => {
+    const fileExtension = file.name.split(".").pop()?.toLowerCase();
+    if (fileExtension !== "csv") {
       alert("Unsupported file format. Please upload a CSV file.");
+      return;
     }
+
+    importCsv(
+      file,
+      version,
+      (newCriteria) => {
+        updateRubric(newCriteria);
+      },
+      (error) => {
+        setErrorMessage(error);
+      },
+    );
   };
 
-  const parseCSV = (file: File) => {
-    Papa.parse(file, {
-      header: false, // keeps the output an array to sync with parsing xlsx files
-      complete: (results) => {
-        console.log("Parsed CSV data:", results.data);
+  const updateRubric = (newCriteria: Criteria[]) => {
+    setRubric((prevRubric) =>
+      prevRubric
+        ? {
+            ...prevRubric,
+            criteria: [...prevRubric.criteria, ...newCriteria],
+            pointsPossible:
+              prevRubric.pointsPossible +
+              newCriteria.reduce((sum, criterion) => sum + criterion.points, 0),
+          }
+        : {
+            id: parseInt(uuid(), 16),
```

Comment:
id is an optional field as it's automatically assigned by Canvas and generating one doesn't really do anything for us

Reply:
so this can be removed

Reply:
also parsing a uuid to an int value isn't super reliable for uniqueness anyways

Reply:
Removed uuid and used placeholder.